custom_components/tahoma/water_heater_devices/domestic_hot_water_production.py

```diff
@@ -87,36 +87,72 @@ async def async_set_temperature(self, **kwargs):
 
     async def async_set_operation_mode(self, operation_mode):
         """Set new target operation mode."""
+        if operation_mode == OverkizCommandParam.BOOST:
+            await self.executor.async_execute_command(
+                OverkizCommand.SET_BOOST_MODE, OverkizCommand.ON
+            )
+            return
+        if operation_mode != OverkizCommandParam.BOOST and self._is_boost_mode_on:
+            await self.executor.async_execute_command(
+                OverkizCommand.SET_BOOST_MODE, OverkizCommand.OFF
+            )
         await self.executor.async_execute_command(
             OverkizCommand.SET_DHW_MODE, OPERATION_MODE_TO_OVERKIZ[operation_mode]
         )
 
     @property
     def is_away_mode_on(self):
         """Return true if away mode is on."""
-        return (
-            self.executor.select_state(OverkizState.CORE_OPERATING_MODE).get(
-                OverkizCommandParam.ABSENCE
+        if self.device.controllable_name == DHWP_TYPE_MURAL:
+            return (
+                self.executor.select_state(OverkizState.CORE_OPERATING_MODE).get(
+                    OverkizCommandParam.ABSENCE
+                )
+                == OverkizCommandParam.ON
+            )
+        if self.device.controllable_name == DHWP_TYPE_CE_FLAT_C2:
+            return (
+                self.executor.select_state(OverkizState.IO_DHW_ABSENCE_MODE)
+                == OverkizCommandParam.ON
+            )
+
+    @property
+    def _is_boost_mode_on(self):
+        """Return true if boost mode is on."""
+        if self.device.controllable_name == DHWP_TYPE_MURAL:
```

Comment:
any better idea than this would be highly appreciated!

Reply:
Away mode works nicely. Boost mode is not working. It doesn't appear in modes menu, and if it try to change mode: manual to eco, for example i got this error:
 `2021-12-22 12:09:29 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [140546044862768] 'super' object has no attribute '_is_boost_mode_on'
Traceback (most recent call last):
  File "/srv/homeassistant/lib/python3.9/site-packages/homeassistant/components/websocket_api/commands.py", line 185, in handle_call_service
    await hass.services.async_call(
  File "/srv/homeassistant/lib/python3.9/site-packages/homeassistant/core.py", line 1495, in async_call
    task.result()
  File "/srv/homeassistant/lib/python3.9/site-packages/homeassistant/core.py", line 1530, in _execute_service
    await handler.job.target(service_call)
  File "/srv/homeassistant/lib/python3.9/site-packages/homeassistant/helpers/entity_component.py", line 209, in handle_service
    await self.hass.helpers.service.entity_service_call(
  File "/srv/homeassistant/lib/python3.9/site-packages/homeassistant/helpers/service.py", line 663, in entity_service_call
    future.result()  # pop exception if have
  File "/srv/homeassistant/lib/python3.9/site-packages/homeassistant/helpers/entity.py", line 896, in async_request_call
    await coro
  File "/srv/homeassistant/lib/python3.9/site-packages/homeassistant/helpers/service.py", line 700, in _handle_entity_call
    await result
  File "/home/homeassistant/.homeassistant/custom_components/tahoma/water_heater_devices/domestic_hot_water_production.py", line 95, in async_set_operation_mode
    if operation_mode != OverkizCommandParam.BOOST and self._is_boost_mode_on:
  File "/home/homeassistant/.homeassistant/custom_components/tahoma/water_heater_devices/domestic_hot_water_production.py", line 123, in _is_boost_mode_on
    return super()._is_boost_mode_on
AttributeError: 'super' object has no attribute '_is_boost_mode_on'
`